### PR TITLE
[4.3] Expose configs, alter PR template

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- The configuration files for ESLint and StyleLint was placed inside the build folder (mainly because those were hidden files). Now both these tools allow the `name.config.json` convention so we can move them back to the root. The build.php was updated to exclude them from the packaging process.
- Similarly, the settings for the localisation of node_modules and the creation of the error pages was also stored in a file named settings.json inside the build folder. The whole section is now in the package.json under the key `joomlaSettings`. All the scripts affected by this change were updated.
- Finally a section before documentation was added to the PR template and looks like:

```js
// Preview starts here
```

### Codestyle conformance
Please make sure that proir to creating a Pull Request you have ran locally the following commands:

- PHP:      `./libraries/vendor/bin/phpcs --extensions=php -p --standard=ruleset.xml .`
- JS:       `npm run lint:js` or for autofixing minor problems `npm run lint:js -- --fix`
- SCSS/CSS: `npm run lint:css` or for autofixing minor problems `npm run lint:css -- --fix`

```js
// Preview ends here
```

### Testing Instructions
Test that all the commands in the package.json still work as expected.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

Documented in the `build/README.md`

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -16,6 +16,14 @@ Pull Request for Issue # .
 
 
 
+### Codestyle conformance
+Please make sure that proir to creating a Pull Request you have ran locally the following commands:
```
Comment (richard67) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Please make sure that prior to creating a Pull Request you have ran locally the following commands:
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -16,6 +16,14 @@ Pull Request for Issue # .
 
 
 
+### Codestyle conformance
+Please make sure that prior to creating a Pull Request you have ran locally the following commands:
```
Comment (brianteeman) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Please make sure that prior to creating a Pull Request you have run locally the following commands:
```

File: build/README.md [hunk 3]
```diff
@@ -67,4 +67,11 @@ Assuming that you are working on the Web Authentication JavaScript, you can run
 
 Once you get your code doing what it is meant to do, make sure that you check you are not breaking any of the Code Style rules by running `npm run lint:css -- --fix` and `npm run lint:js -- --fix` (the `-- --fix` will try to fix anything that's not trivial).
 
+## IDE tools
+IDEs like PHPStorm and VSCode have tools also for compiling/linting. At the moment Joomla exposes the following configurations:
```
Comment (brianteeman) [inline build/README.md]:
```suggestion
IDEs like PHPStorm and VSCode also have tools for compiling/linting. At the moment Joomla exposes the following configurations:
```

**Comment by HLeithner** [comment]
can your solve the conflicts? @Fedik any objections?

**Comment by Fedik** [comment]
No objections ;)

**Pull request closed without merging**